lib/globus_client.rb

```diff
@@ -59,70 +69,168 @@ def default_auth_url
 
   attr_accessor :config
 
+  # Send an authenticated GET request
+  # @param base_url [String] the base URL of the Globus API
+  # @param path [String] the path to the Globus API request
+  # @param params [Hash] params to get to the API
+  def get(base_url:, path:, params: {}, content_type: nil)
+    response = with_token_refresh_when_unauthorized do
+      connection(base_url).get(path, params) do |request|
+        request.headers["Authorization"] = "Bearer #{config.token}"
+        request.headers["Content-Type"] = content_type if content_type
+      end
+    end
+
+    UnexpectedResponse.call(response) unless response.success?
+
+    return nil if response.body.blank?
+
+    JSON.parse(response.body)
+  end
+
+  # Send an authenticated POST request
+  # @param base_url [String] the base URL of the Globus API
+  # @param path [String] the path to the Globus API request
+  # @param expected_response [#call] an expected response handler to allow short-circuiting the unexpected response
+  def post(base_url:, path:, body:, expected_response: ->(resp) { false })
+    response = with_token_refresh_when_unauthorized do
+      connection(base_url) { |conn| conn.request :retry, HTTP_RETRY_OPTIONS }
+        .post(path) do |request|
+        request.headers["Authorization"] = "Bearer #{config.token}"
+        request.headers["Content-Type"] = "application/json"
+        request.body = body.to_json
+      end
+    end
+
+    UnexpectedResponse.call(response) unless response.success? || expected_response.call(response)
+
+    return nil if response.body.blank?
+
+    JSON.parse(response.body)
+  end
+
+  # Send an authenticated PUT request
+  # @param base_url [String] the base URL of the Globus API
+  # @param path [String] the path to the Globus API request
+  def put(base_url:, path:, body:)
```

Comment:
since the PR description indicates integration tests passed, i'm wondering whether i'm mis-understanding something, or if this is just something the integration tests don't cover (since i'm not super familiar with the ins-and-outs of the interactions between globus and H2).

Reply:
Yep, good catch. There's only use of `PUT` in the client, and I suspect the API test doesn't cover it. I'll see if I can add that as part of this PR.